dart/lib/src/sentry.dart

```diff
@@ -34,6 +49,38 @@ class Sentry {
     await _init(options);
   }
 
+  static Future<void> _initDefaultValues(
+    SentryOptions options,
+    AppRunner appRunner,
+  ) async {
+    // if no environment is set, we set 'production' by default, but if we know it's
+    // a non-release build, or the SENTRY_ENVIRONMENT is set, we read from it.
+    if (const bool.hasEnvironment('SENTRY_ENVIRONMENT') || !isReleaseMode) {
+      options.environment = const String.fromEnvironment(
+        'SENTRY_ENVIRONMENT',
+        defaultValue: 'debug',
+      );
+    }
+
+    // if the SENTRY_DSN is set, we read from it.
+    options.dsn = const bool.hasEnvironment('SENTRY_DSN')
+        ? const String.fromEnvironment('SENTRY_DSN')
+        : options.dsn;
+
+    // Throws when running on the browser
+    if (!isWeb) {
+      // catch any errors that may occur within the entry function, main()
+      // in the ‘root zone’ where all Dart programs start
+      options.addIntegration(isolateErrorIntegration);
+    }
+
+    // finally the runZonedGuarded, catch any errors in Dart code running
+    // ‘outside’ the Flutter framework
+    if (appRunner != null) {
+      options.addIntegration(runZonedGuardedIntegration(appRunner));
+    }
```

Comment:
One of the crucial things of this feature was to preserve the `ordering of the execution` of the integration and this depends on the order they were added to the `options.integrations` list.

I see now that `runZonedGuardedIntegration` is gonna be executed before `flutterErrorIntegration` and this would cause Flutter errors to not be reported on SDK init.

Maybe `Sentry.init` should accept an optional parameter which is the Integration that should be added before `runZonedGuardedIntegration`, or reordering the integrations list on `SentryFlutter.init`, not sure which one is more viable, but the first one seems less flakey.

Reply:
✅ added a positional `initialIntegrations` to `Sentry.init`